akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/ActiveEventProducer.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.grpc.producer.scaladsl
+
+import akka.Done
+import akka.NotUsed
+import akka.actor.typed.ActorSystem
+import akka.grpc.GrpcClientSettings
+import akka.grpc.scaladsl.Metadata
+import akka.grpc.scaladsl.MetadataBuilder
+import akka.persistence.query.typed.EventEnvelope
+import akka.projection.ProjectionContext
+import akka.projection.grpc.internal.EventPusher
+import akka.projection.grpc.internal.proto.EventConsumerServiceClient
+import akka.projection.grpc.producer.scaladsl.EventProducer.EventProducerSource
+import akka.stream.scaladsl.FlowWithContext
+
+object ActiveEventProducer {
+
+  def apply[Event](
+      originId: String,
+      eventProducerSource: EventProducerSource,
+      connectionMetadata: Metadata,
+      grpcHost: String,
+      grpcPort: Int): ActiveEventProducer[Event] =
+    new ActiveEventProducer[Event](originId, eventProducerSource, Some(connectionMetadata), grpcHost, grpcPort)
+
+  def apply[Event](
+      originId: String,
+      eventProducerSource: EventProducerSource,
+      grpcHost: String,
+      grpcPort: Int): ActiveEventProducer[Event] =
+    new ActiveEventProducer[Event](originId, eventProducerSource, None, grpcHost, grpcPort)
+}
+
+final class ActiveEventProducer[Event](
+    val originId: String,
+    val eventProducerSource: EventProducerSource,
+    val connectionMetadata: Option[Metadata],
+    val grpcHost: String,
+    val grpcPort: Int) {
+
+  def handler()(implicit system: ActorSystem[_])
+      : FlowWithContext[EventEnvelope[Event], ProjectionContext, Done, ProjectionContext, NotUsed] = {
+    // FIXME gprc client config - use stream id to look up block? Use host to lookup port (normal gRPC client config)? Something else?
+    val eventConsumerClient = EventConsumerServiceClient(
+      GrpcClientSettings.connectToServiceAt(grpcHost, grpcPort).withTls(false))
```

Comment:
yes, I guess we should have something similar to how grpc client is provided in GrpcReadJournal

also note `channelBuilderOverrides` there

Reply:
I pushed taking the gRPC client settings from the user while you were reviewing, I think that is better because that way finding the place to push to can be whatever kind of lookup, from config etc. that Akka gRPC supports.

Reply:
Still missing a way where it can be defined in config per stream id, with a grpc client block.

Reply:
I think we'll skip any pre-defined config structure and leave up to the user to read GrpcClientSettings from config somewhere that suits them.